src/ucp/rndv/rndv.c

```diff
@@ -721,23 +724,70 @@ static void ucp_rndv_req_init_zcopy_lane_map(ucp_request_t *rndv_req,
         }
     }
 
-    if (ucs_popcount(lane_map) > 1) {
-        /* remove lanes if bandwidth is too less compare to best lane */
-        ucs_for_each_bit(lane_idx, lane_map) {
+    if (ucs_popcount(lane_map) <= 1) {
+        goto out;
+    }
+
+    /* Remove lanes if bandwidth is too small compared to the best lane */
+    ucs_for_each_bit(lane_idx, lane_map) {
+        ucs_assert(lane_idx < UCP_MAX_LANES);
+        lane       = lanes[lane_idx];
+        rsc_index  = ep_config->key.lanes[lane].rsc_index;
+        iface_attr = ucp_worker_iface_get_attr(ep->worker, rsc_index);
+        lane_bw    = ucp_tl_iface_bandwidth(context, &iface_attr->bandwidth);
+
+        if ((lane_bw / max_lane_bw) <
+            (1. / context->config.ext.multi_lane_max_ratio)) {
+            lane_map                                &= ~UCS_BIT(lane_idx);
+            rndv_req->send.rndv.rkey_index[lane_idx] = UCP_NULL_RESOURCE;
+            continue;
+        }
+
+        if (UCS_BIT(ep_config->md_index[lane]) & md_map) {
+            exclude_map |= UCS_BIT(lane_idx);
+        } else {
+            md_map |= UCS_BIT(ep_config->md_index[lane]);
+        }
+    }
+
+    chunk_count = ucs_max(rndv_req->send.length /
+                              context->config.ext.min_rndv_chunk_size,
+                          1);
+
+    /* If there are more lanes than chunks of the mimimal size, we have to
+     * exclude some lanes, so that there will be no more that 1 lane per chunk.
```

Comment:
i meant it like here https://en.wiktionary.org/wiki/so_that

Reply:
sorry I wasn't clear - didn't noticed usage of `that` twice here:
```suggestion
     * exclude some lanes, so that there will be no more than 1 lane per chunk.
```